packages/devtools_app/lib/src/screens/memory/panes/chart/legend.dart

```diff
@@ -14,6 +14,102 @@ import 'memory_charts.dart';
 import 'memory_events_pane.dart';
 import 'memory_vm_chart.dart';
 
+class MemoryChartWidget extends StatelessWidget {
+  const MemoryChartWidget({
+    super.key,
+    required this.isAndroidVisible,
+    required this.chartController,
+  });
+
+  final bool isAndroidVisible;
+  final MemoryChartPaneController chartController;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final colorScheme = theme.colorScheme;
+    final legendHeading = theme.hoverTextStyle;
+
+    final legendRows = <Widget>[];
+
+    final events = eventLegendContent(colorScheme.isLight);
+    legendRows.add(
+      Container(
+        padding: legendTitlePadding,
+        child: Text('Events Legend', style: legendHeading),
+      ),
+    );
+
+    var iterator = events.entries.iterator;
+    while (iterator.moveNext()) {
```

Comment:
nit: use a for loop instead

Reply:
not in this loop as it is doing iterator.moveNext() in the body